main_test.go

```diff
@@ -0,0 +1,158 @@
+}
+
+func (s *testsession) AddHandler(handler interface{}) func() {
+	s.handler = handler.(func(s *discordgo.Session, m *discordgo.MessageCreate))
+	return func() {}
+}
+
+func (s *testsession) Channel(channelID string) (*discordgo.Channel, error) {
+	panic("Channel not implemented")
+}
+
+func (s *testsession) ChannelMessage(channelID, messageID string) (*discordgo.Message, error) {
+	panic("ChannelMessage not implemented")
+}
+
+func (s *testsession) ChannelMessages(channelID string, limit int, beforeID, afterID, aroundID string) (
+	[]*discordgo.Message, error) {
+	panic("ChannelMessages not implemented")
+}
+
+func (s *testsession) ChannelMessagesPinned(channelID string) ([]*discordgo.Message, error) {
+	panic("ChannelMessagesPinned not implemented")
+}
+
+func (s *testsession) ChannelTyping(channelID string) (err error) {
+	panic("ChannelTyping not implemented")
+}
+
+func (s *testsession) ChannelMessageSend(channelID string, content string) (*discordgo.Message, error) {
+	b, ok := s.messages[channelID]
+	if !ok {
+		b = new(strings.Builder)
+		s.messages[channelID] = b
+	}
+	b.WriteString(content)
+	b.WriteByte('\n')
+	s.counter++
+	return newMsg(content, channelID, strconv.Itoa(s.counter)), nil
+}
+
+func (s *testsession) ChannelMessageSendEmbeds(channelID string, embeds []*discordgo.MessageEmbed) (
+	*discordgo.Message, error) {
+	panic("ChannelMessageSendEmbeds not implemented")
+}
+
+func (s *testsession) ChannelMessageSendTTS(channelID string, content string) (*discordgo.Message, error) {
+	panic("ChannelMessageSendTTS not implemented")
+}
+
+func (s *testsession) ChannelFileSend(channelID, name string, r io.Reader) (*discordgo.Message, error) {
+	panic("ChannelFileSend not implemented")
+}
+
+func (s *testsession) ChannelFileSendWithMessage(channelID, content string, name string, r io.Reader) (
+	*discordgo.Message, error) {
+	panic("ChannelFileSendWithMessage not implemented")
+}
+
+func (s *testsession) ChannelMessageEdit(channelID, messageID, content string) (*discordgo.Message, error) {
+	panic("ChannelMessageEdit not implemented")
+}
+
+func (s *testsession) ChannelMessageEditEmbeds(channelID, messageID string, embeds []*discordgo.MessageEmbed) (
+	*discordgo.Message, error) {
+	panic("ChannelMessageEditEmbeds not implemented")
+}
+
+func (s *testsession) ChannelMessageDelete(channelID, messageID string) error {
+	panic("ChannelMessageDelete not implemented")
+}
+
+func (s *testsession) ChannelMessagesBulkDelete(channelID string, messages []string) error {
+	panic("ChannelMessagesBulkDelete not implemented")
+}
+
+func (s *testsession) ChannelMessagePin(channelID, messageID string) error {
+	panic("ChannelMessagePin not implemented")
+}
+
+func (s *testsession) ChannelMessageUnpin(channelID, messageID string) error {
+	panic("ChannelMessageUnpin not implemented")
+}
+
+func (s *testsession) UpdateGameStatus(idle int, name string) error {
+	panic("UpdateGameStatus not implemented")
```

Comment:
Hmm, so how does play_test work when a mock impl of this isn't available?

Reply:
This is for the top level eelbot tests (which tests registering commands and replies). Command specific tests (like the play command test) are in the `commands` subfolder, which does have an implementation for this.

Reply:
Where? I don't see a stub implementation of this in https://github.com/emseers/eelbot/blob/a03ef59768e98830d4a632b32de5565aa368f1dc/commands/play_test.go

Reply:
It's here: https://github.com/emseers/eelbot/blob/a03ef59768e98830d4a632b32de5565aa368f1dc/commands/main_test.go#L204